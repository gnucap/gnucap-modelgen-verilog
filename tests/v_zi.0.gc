attach ../vams/vsine.so


verilog
parameter T = {3.141592 / 4.}

// H(z) = 1/ (z * z * (1-z)) = 1/(zz-zzz)
va_zi #(.d0(1.), .n0(0.), .n2(1.) .n3(-1.), .T(T*.5), .tau(T*.1)) b(0,out1,in,0);
va_zi #(.d0(1.), .xr2(1.), .T(T*.5), .tau(T*.1)) b(0,out2,in,0);
va_zi #(.d0(1.), .n0(1.), .T(1.)) b(0,out3,in,0);
vsine #(.ampl(1), .freq(1/2/3.1415)) v1(in, 0);

list

print tran v(in) v(out*) control(0)
tran 10 basic
//
print ac vm(out*) vp(out*)
ac .05 0.2 * 1.1

list
status notime
