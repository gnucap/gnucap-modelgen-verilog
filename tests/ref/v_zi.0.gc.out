parameter T = {3.141592 / 4.} 
// H(z) = 1/ (z * z * (1-z)) = 1/(zz-zzz) 
va_zi #(.n0(0.),.n2(1.),.n3(-1.),.d0(1.)) b (.out0(0),.out1(out1),.in0(in),.in1(0));
va_zi #(.xr2(1.),.d0(1.)) b (.out0(0),.out1(out2),.in0(in),.in1(0));
va_zi #(.n0(1.),.d0(1.)) b (.out0(0),.out1(out3),.in0(in),.in1(0));
vsine #(.freq(1/2/3.1415),.ampl(1)) v1 (.p(in),.n(0));
#Time       v(in)      v(out1)    v(out2)    v(out3)    control(0)
 0.         0.         0.         0.         0.         1.        
 0.07854    0.078461   0.         0.         0.         7.        
 0.1        0.099836   0.         0.         0.         7.        
 0.3927     0.38269    0.         0.         0.         2.        
 0.47124    0.454      0.         0.         0.         7.        
 0.7854     0.70712    0.         0.         0.         2.        
 0.86394    0.76042    0.         0.         0.         7.        
 1.         0.84149    0.         0.         0.         2.        
 1.1        0.89122    0.         0.         0.84149    7.        
 1.1781     0.92389    0.         0.         0.84149    2.        
 1.2566     0.95107    0.38269    0.38269    0.84149    7.        
 1.5708     1.         0.38269    0.38269    0.84149    2.        
 1.6493     0.99691    0.32443    0.32443    0.84149    7.        
 1.9635     0.92386    0.32443    0.32443    0.84149    2.        
 2.         0.90927    0.27439    0.27439    0.84149    2.        
 2.042      0.89098    0.21677    0.21677    0.86998    7.        
 2.1        0.86318    0.21677    0.21677    0.90927    7.        
 2.3562     0.70706    0.21677    0.21677    0.90927    2.        
 2.4347     0.64939    0.076107   0.076107   0.90927    7.        
 2.7489     0.38261    0.076107   0.076107   0.90927    2.        
 2.8274     0.30894   -0.076142  -0.076142   0.90927    7.        
 3.         0.14103   -0.076142  -0.076142   0.90927    2.        
 3.1        0.041489  -0.076142  -0.076142   0.14103    7.        
 3.1416    -92.003E-6 -0.076142  -0.076142   0.14103    2.        
 3.2201    -0.078553  -0.2168    -0.2168     0.14103    7.        
 3.5343    -0.38278   -0.2168    -0.2168     0.14103    2.        
 3.6128    -0.45408   -0.32445   -0.32445    0.14103    7.        
 3.927     -0.70719   -0.32445   -0.32445    0.14103    2.        
 4.        -0.75688   -0.3786    -0.3786     0.14103    2.        
 4.0055    -0.76048   -0.3827    -0.3827     0.09138    7.        
 4.1       -0.81835   -0.3827    -0.3827    -0.75688    7.        
 4.3197    -0.92393   -0.3827    -0.3827    -0.75688    2.        
 4.3982    -0.9511    -0.38269   -0.38269   -0.75688    7.        
 4.7124    -1.        -0.38269   -0.38269   -0.75688    2.        
 4.7909    -0.99691   -0.32441   -0.32441   -0.75688    7.        
 5.        -0.95888   -0.32441   -0.32441   -0.75688    2.        
 5.1       -0.92576   -0.32441   -0.32441   -0.95888    7.        
 5.1051    -0.92382   -0.32441   -0.32441   -0.95888    2.        
 5.1836    -0.89094   -0.21674   -0.21674   -0.95888    7.        
 5.4978    -0.70699   -0.21674   -0.21674   -0.95888    2.        
 5.5763    -0.64932   -0.076072  -0.076072  -0.95888    7.        
 5.8905    -0.38252   -0.076072  -0.076072  -0.95888    2.        
 5.969     -0.30885    0.076178   0.076178  -0.95888    7.        
 6.        -0.27925    0.076178   0.076178  -0.95888    2.        
 6.1       -0.18199    0.076178   0.076178  -0.27925    7.        
 6.2832     184.01E-6  0.076178   0.076178  -0.27925    2.        
 6.3617     0.078645   0.21683    0.21683   -0.27925    7.        
 6.6759     0.38286    0.21683    0.21683   -0.27925    2.        
 6.7544     0.45417    0.32447    0.32447   -0.27925    7.        
 7.         0.65714    0.32447    0.32447   -0.27925    2.        
 7.0686     0.70725    0.32447    0.32447    0.36295    2.        
 7.1        0.72911    0.34777    0.34777    0.65714    7.        
 7.1471     0.76054    0.38271    0.38271    0.65714    7.        
 7.4613     0.92396    0.38271    0.38271    0.65714    2.        
 7.5398     0.95112    0.38268    0.38268    0.65714    7.        
 7.854      1.         0.38268    0.38268    0.65714    2.        
 7.9325     0.9969     0.32439    0.32439    0.65714    7.        
 8.         0.98932    0.32439    0.32439    0.65714    2.        
 8.1        0.96983    0.32439    0.32439    0.98932    7.        
 8.2467     0.92379    0.32439    0.32439    0.98932    2.        
 8.3252     0.8909     0.21671    0.21671    0.98932    7.        
 8.6394     0.70693    0.21671    0.21671    0.98932    2.        
 8.7179     0.64925    0.076037   0.076037   0.98932    7.        
 9.         0.41188    0.076037   0.076037   0.98932    2.        
 9.0321     0.38244    0.076037   0.076037   0.8041     2.        
 9.1        0.31884   -0.055632  -0.055632   0.41188    7.        
 9.1106     0.30876   -0.076213  -0.076213   0.41188    7.        
 9.4248    -276.01E-6 -0.076213  -0.076213   0.41188    2.        
 9.5033    -0.078737  -0.21686   -0.21686    0.41188    7.        
 9.8175    -0.38295   -0.21686   -0.21686    0.41188    2.        
 9.896     -0.45425   -0.32449   -0.32449    0.41188    7.        
 10.       -0.54427   -0.32449   -0.32449    0.41188    1.        
#Freq       vm(out1)   vm(out2)   vm(out3)   vp(out1)   vp(out2)   vp(out3)  
 0.05       0.12329    0.12329    1.         71.622     71.622    -0.9       
 0.055      0.1356     0.1356     1.         69.784     69.784    -0.99      
 0.0605     0.14914    0.14914    1.         67.762     67.762    -1.089     
 0.06655    0.16402    0.16402    1.         65.538     65.538    -1.1979    
 0.073205   0.18038    0.18038    1.         63.092     63.092    -1.3177    
 0.080526   0.19836    0.19836    1.         60.402     60.402    -1.4495    
 0.088578   0.21812    0.21812    1.         57.442     57.442    -1.5944    
 0.097436   0.23983    0.23983    1.         54.186     54.186    -1.7538    
 0.10718    0.26368    0.26368    1.         50.604     50.604    -1.9292    
 0.1179     0.28988    0.28988    1.         46.665     46.665    -2.1222    
 0.12969    0.31863    0.31863    1.         42.331     42.331    -2.3344    
 0.14266    0.35017    0.35017    1.         37.565     37.565    -2.5678    
 0.15692    0.38477    0.38477    1.         32.321     32.321    -2.8246    
 0.17261    0.4227     0.4227     1.         26.553     26.553    -3.107     
 0.18987    0.46422    0.46422    1.         20.208     20.208    -3.4177    
 0.20886    0.50966    0.50966    1.         13.229     13.229    -3.7595    
parameter T = {3.141592 / 4.} 
// H(z) = 1/ (z * z * (1-z)) = 1/(zz-zzz) 
va_zi #(.n0(0.),.n2(1.),.n3(-1.),.d0(1.)) b (.out0(0),.out1(out1),.in0(in),.in1(0));
va_zi #(.n0( 0.),.n1( 0.),.n2( 1.),.n3(-1.),.d0(1.)) b (.out0(0),.out1(out2),.in0(in),.in1(0));
va_zi #(.n0(1.),.d0(1.)) b (.out0(0),.out1(out3),.in0(in),.in1(0));
vsine #(.freq(1/2/3.1415),.ampl(1)) v1 (.p(in),.n(0));
// 
Gnucap   System status
iterations: op=0, dc=0, tran=216, fourier=0, total=304
transient timesteps: accepted=72, rejected=0, total=72
nodes: user=4, subckt=0, model=1, total=5
dctran density=44.0%, ac density=52.0%
